airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_jsonl_parser.py

```diff
@@ -33,3 +35,66 @@ def test_type_mapping(record: Dict[str, Any], expected_schema: Mapping[str, str]
             JsonlParser().infer_schema_for_record(record)
     else:
         assert JsonlParser.infer_schema_for_record(record) == expected_schema
+
+
+JSONL_CONTENT_WITHOUT_MULTILINE_JSON_OBJECTS = [
+    b'{"a": 1, "b": "1"}',
+    b'{"a": 2, "b": "2"}',
+]
+JSONL_CONTENT_WITH_MULTILINE_JSON_OBJECTS = [
+    b'{',
+    b'  "a": 1,',
+    b'  "b": "1"',
+    b'}',
+    b'{',
+    b'  "a": 2,',
+    b'  "b": "2"',
+    b'}',
+]
+INVALID_JSON_CONTENT = [
+    b'{',
+    b'  "a": 1,',
+    b'  "b": "1"',
+    b'{',
+    b'  "a": 2,',
+    b'  "b": "2"',
+    b'}',
+]
+
+
+def test_given_one_json_per_line_when_parse_records_then_return_records() -> None:
+    stream_reader = MagicMock()
+    stream_reader.open_file.return_value.__enter__.return_value = JSONL_CONTENT_WITHOUT_MULTILINE_JSON_OBJECTS
+
+    records = list(JsonlParser().parse_records(Mock(), Mock(), stream_reader, Mock()))
+
+    assert records == [{"a": 1, "b": "1"}, {"a": 2, "b": "2"}]
+
+
+def test_given_multiline_json_object_when_parse_records_then_return_records() -> None:
```

Comment:
this test should verify we don't log a warning

Reply:
Did you mean that we should log a warning? `test_given_multiline_json_object_when_parse_records_then_log_once_one_record_yielded` validates that. I can check that `test_given_one_json_per_line_when_parse_records_then_return_records` does not log a warning for extra security

Reply:
I think I derped and commented on the wrong line. +1 for verifying we do not log a warning when only parsing single line records